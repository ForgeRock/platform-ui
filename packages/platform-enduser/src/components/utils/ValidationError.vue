<template>
  <div
    v-show="validatorErrors.has(fieldName)"
    class="fr-validation-requirements text-left">
    <span class="text-danger">
      {{ validatorErrors.first(fieldName) }}
    </span>
  </div>
</template>

<script>
/**
 * @description Component for displaying error message for form fields
 *
 * */
export default {
  name: 'ValidationError',
  props: {
    validatorErrors: {
      type: Object,
      default: () => ({}),
    },
    fieldName: {
      type: String,
      default: () => '',
    },
  },
};
</script>

<style lang="scss" scoped></style>
