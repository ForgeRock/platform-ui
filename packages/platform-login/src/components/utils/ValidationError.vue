<template>
  <div
    v-show="validatorErrors.length"
    class="fr-validation-requirements text-left">
    <p
      v-for="(error) in validatorErrors"
      :key="error"
      class="text-danger mb-0">
      {{ error }}
    </p>
  </div>
</template>

<script>
// TODO Add to components
/**
 * @description Component for displaying error message for form fields
 *
 * */
export default {
  name: 'ValidationError',
  props: {
    validatorErrors: {
      type: Array,
      default: () => [],
    },
    fieldName: {
      type: String,
      default: '',
    },
  },
};
</script>

<style lang="scss" scoped></style>
