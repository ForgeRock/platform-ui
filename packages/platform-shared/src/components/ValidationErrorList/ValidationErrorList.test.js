/**
 * Copyright (c) 2019-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import BootstrapVue from 'bootstrap-vue';
import { noop } from 'lodash';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import ValidationError from './index';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('ValidationError.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(ValidationError, {
      mocks: {
        $t: () => {},
      },
      propsData: {
        localVue,
        validatorErrors: {
          has: noop,
          first: noop,
        },
        fieldName: 'test',
      },
    });
  });

  it('Validation Error component loaded', () => {
    expect(wrapper.name()).toEqual('ValidationErrorList');
  });
});
