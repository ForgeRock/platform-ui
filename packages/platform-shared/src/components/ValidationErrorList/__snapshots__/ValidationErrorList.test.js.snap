// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationError.vue Validation Error component loaded 1`] = `
<div class="fr-validation-requirements text-left" localvue="function VueComponent (options) {
      this._init(options);
    }"><span class="text-danger">

  </span></div>
`;
